pkg/api/customization/oci/handler.go

```diff
@@ -0,0 +1,217 @@
+	if err != nil {
+		util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+		return
+	}
+
+	provider := common.NewRawConfigurationProvider(creds.Tenancy, creds.User, creds.Region, creds.FingerPrint,
+		creds.PrivateKey, &creds.PrivateKeyPassphrase)
+
+	resourceType := mux.Vars(req)["resource"]
+
+	var serialized []byte
+
+	switch resourceType {
+	case "vcns":
+		if serialized, err = processVcns(provider, creds.Compartment); err != nil {
+			logrus.Debugf("error processing VCNs: %v", err)
+			util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "okeVersions":
+		if serialized, err = processOkeVersions(provider); err != nil {
+			logrus.Debugf("error processing OKE versions: %v", err)
+			util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "availabilityDomains":
+		if serialized, err = processAvailabilityDomains(provider, creds.Compartment); err != nil {
+			logrus.Debugf("error processing ADs: %v", err)
+			util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "regions":
+		if serialized, err = processRegions(provider, creds.Tenancy); err != nil {
+			logrus.Debugf("error processing regions: %v", err)
+			util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "nodeShapes":
+		if serialized, err = processNodeShapes(provider, creds.Compartment); err != nil {
+			util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "nodeImages":
+		if serialized, err = processImages(provider, creds.Compartment); err != nil {
+			util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	case "nodeOkeImages":
+		if serialized, err = processNodeOkeImages(provider); err != nil {
+			util.ReturnHTTPError(writer, req, httperror.ServerError.Status, err.Error())
+			return
+		}
+		writer.Write(serialized)
+	default:
+		writer.WriteHeader(http.StatusInternalServerError)
+		util.ReturnHTTPError(writer, req, httperror.ServerError.Status, "invalid endpoint "+resourceType)
+	}
+}
+
+// extractCreds attempts to extract the credentials from a given request either from the body or cloud credentials.
+func (handler *Handler) extractCreds(writer http.ResponseWriter, req *http.Request, creds *Credentials) error {
+
+	credID := req.URL.Query().Get("cloudCredentialId")
+	if credID != "" {
+		ns, name := ref.Parse(credID)
+		if ns == "" || name == "" {
+			writer.WriteHeader(http.StatusUnprocessableEntity)
+			logrus.Debugf("invalid cloud credential ID %s", credID)
+			return fmt.Errorf("invalid cloud credential ID %s", credID)
+		}
+
+		cc, err := handler.secretsLister.Get(namespace.GlobalNamespace, name)
```

Comment:
There needs to be a check here that validates the user sending the request has access to the cloud credential being referenced.

Reply:
This should follow the same pattern the vspehere handler is doing

Reply:
This is complete.